extensions/common/vault/vault-azure/src/test/java/org/eclipse/edc/vault/azure/AzureVaultExtensionTest.java

```diff
@@ -8,93 +8,59 @@
- *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - improvements
+ *       Amadeus - initial API and implementation
  *
  */
 
 package org.eclipse.edc.vault.azure;
 
-import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.junit.extensions.DependencyInjectionExtension;
+import org.eclipse.edc.spi.security.CertificateResolver;
+import org.eclipse.edc.spi.security.PrivateKeyResolver;
+import org.eclipse.edc.spi.security.Vault;
+import org.eclipse.edc.spi.security.VaultCertificateResolver;
+import org.eclipse.edc.spi.security.VaultPrivateKeyResolver;
 import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.edc.spi.system.injection.ObjectFactory;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.mockito.MockedStatic;
+import org.junit.jupiter.api.extension.ExtendWith;
 
-import java.util.UUID;
+import static org.assertj.core.api.Assertions.assertThat;
 
-import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.mockStatic;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.when;
+@ExtendWith(DependencyInjectionExtension.class)
+public class AzureVaultExtensionTest {
 
-class AzureVaultExtensionTest {
+    private static final String VAULT_NAME = "aVault";
+    private static final String VAULT_NAME_SETTING = "edc.vault.name";
 
-    private static final String CLIENT_ID = UUID.randomUUID().toString();
-    private static final String TENANT_ID = UUID.randomUUID().toString();
-    private static final String KEYVAULT_NAME = UUID.randomUUID().toString();
-    private static final String CLIENT_SECRET = UUID.randomUUID().toString();
-    private static final String CERTIFICATE_PATH = UUID.randomUUID().toString();
-
-    private final Monitor monitor = mock(Monitor.class);
-    private final ServiceExtensionContext context = mock(ServiceExtensionContext.class);
-
-    private final AzureVaultExtension extension = new AzureVaultExtension();
+    private ServiceExtensionContext context;
+    private AzureVaultExtension extension;
 
     @BeforeEach
-    public void setUp() {
-        when(context.getSetting("edc.vault.clientid", null)).thenReturn(CLIENT_ID);
-        when(context.getSetting("edc.vault.name", null)).thenReturn(KEYVAULT_NAME);
-        when(context.getSetting("edc.vault.tenantid", null)).thenReturn(TENANT_ID);
-        when(context.getMonitor()).thenReturn(monitor);
-    }
-
-    @Test
-    void neitherSecretOrCertificateProvided_shouldThrowException() {
-        assertThrows(AzureVaultException.class, () -> extension.initialize(context));
+    void setUp(ServiceExtensionContext context, ObjectFactory factory) {
+        this.context = context;
+        extension = factory.constructInstance(AzureVaultExtension.class);
```

Comment:
no need to explicitly instantiate the extension, you can inject the `AzureVaultExtension` directly in the tests, same for the context

Reply:
They will depend on the method you choose to authenticate, as described in the Azure spec. For some, a secret is needed, for other it isn't.
I can reference the Azure doc in the doc tough, is it ok?

Reply:
sure it is

Reply:
I just added a readme file